jlm/llvm/Makefile.sub

```diff
@@ -66,71 +66,78 @@ libllvm_SOURCES = \
     jlm/llvm/opt/unroll.cpp \
 
 libllvm_HEADERS = \
-	jlm/llvm/opt/unroll.hpp \
-	jlm/llvm/opt/DeadNodeElimination.hpp \
-	jlm/llvm/opt/inlining.hpp \
-	jlm/llvm/opt/cne.hpp \
-	jlm/llvm/opt/push.hpp \
-	jlm/llvm/opt/alias-analyses/DifferencePropagation.hpp \
-	jlm/llvm/opt/alias-analyses/LazyCycleDetection.hpp \
-	jlm/llvm/opt/alias-analyses/MemoryNodeProvider.hpp \
-	jlm/llvm/opt/alias-analyses/OnlineCycleDetection.hpp \
-	jlm/llvm/opt/alias-analyses/Optimization.hpp \
-	jlm/llvm/opt/alias-analyses/MemoryStateEncoder.hpp \
-	jlm/llvm/opt/alias-analyses/Steensgaard.hpp \
-	jlm/llvm/opt/alias-analyses/AgnosticMemoryNodeProvider.hpp \
-	jlm/llvm/opt/alias-analyses/RegionAwareMemoryNodeProvider.hpp \
-	jlm/llvm/opt/alias-analyses/PointsToGraph.hpp \
-	jlm/llvm/opt/alias-analyses/AliasAnalysis.hpp \
-	jlm/llvm/opt/pull.hpp \
-	jlm/llvm/opt/optimization.hpp \
-	jlm/llvm/opt/reduction.hpp \
-	jlm/llvm/opt/InvariantValueRedirection.hpp \
-	jlm/llvm/opt/inversion.hpp \
-	jlm/llvm/opt/RvsdgTreePrinter.hpp \
-	jlm/llvm/frontend/LlvmModuleConversion.hpp \
-	jlm/llvm/frontend/LlvmTypeConversion.hpp \
-	jlm/llvm/frontend/ControlFlowRestructuring.hpp \
-	jlm/llvm/frontend/LlvmConversionContext.hpp \
-	jlm/llvm/frontend/LlvmInstructionConversion.hpp \
-	jlm/llvm/frontend/InterProceduralGraphConversion.hpp \
-	jlm/llvm/ir/operators.hpp \
-	jlm/llvm/ir/ipgraph-module.hpp \
-	jlm/llvm/ir/RvsdgModule.hpp \
-	jlm/llvm/ir/linkage.hpp \
-	jlm/llvm/ir/Annotation.hpp \
-	jlm/llvm/ir/attribute.hpp \
-	jlm/llvm/ir/tac.hpp \
-	jlm/llvm/ir/domtree.hpp \
-	jlm/llvm/ir/cfg-node.hpp \
-	jlm/llvm/ir/ipgraph.hpp \
-	jlm/llvm/ir/cfg.hpp \
-	jlm/llvm/ir/ssa.hpp \
-	jlm/llvm/ir/variable.hpp \
-	jlm/llvm/ir/basic-block.hpp \
-	jlm/llvm/ir/print.hpp \
-	jlm/llvm/ir/types.hpp \
-	jlm/llvm/ir/cfg-structure.hpp \
-	jlm/llvm/ir/operators/operators.hpp \
-	jlm/llvm/ir/operators/Phi.hpp \
-	jlm/llvm/ir/operators/Load.hpp \
-	jlm/llvm/ir/operators/MemCpy.hpp \
-	jlm/llvm/ir/operators/MemoryStateOperations.hpp \
-	jlm/llvm/ir/operators/GetElementPtr.hpp \
-	jlm/llvm/ir/operators/delta.hpp \
-	jlm/llvm/ir/operators/Store.hpp \
-	jlm/llvm/ir/operators/alloca.hpp \
-	jlm/llvm/ir/operators/call.hpp \
-	jlm/llvm/ir/operators/sext.hpp \
-	jlm/llvm/ir/operators/lambda.hpp \
-	jlm/llvm/ir/aggregation.hpp \
-	jlm/llvm/backend/dot/DotWriter.hpp \
-	jlm/llvm/backend/rvsdg2jlm/rvsdg2jlm.hpp \
-	jlm/llvm/backend/rvsdg2jlm/context.hpp \
-	jlm/llvm/backend/jlm2llvm/jlm2llvm.hpp \
-	jlm/llvm/backend/jlm2llvm/type.hpp \
-	jlm/llvm/backend/jlm2llvm/instruction.hpp \
-	jlm/llvm/backend/jlm2llvm/context.hpp \
+    jlm/llvm/backend/dot/DotWriter.hpp \
```

Comment:
Do we need to list them individually or could we simply use
jlm/llvm/backend/dot/*.hpp

Reply:
I do not know. @caleridas Could you help me out here, please?

Reply:
Neither I am unsure whether using wildcards here would be a good idea....

Reply:
would prefer to list them explicitly, and certainly a build rule to check for consistency would be in order -- can take a look at that

using wildcards... instinctively I would prefer not to as it also tends to cause problems with spurious dependencies e.g. with files accidentally left in the build tree -- also, when "installing" (=exporting to other projects) files, it is essential that the correct files are selected